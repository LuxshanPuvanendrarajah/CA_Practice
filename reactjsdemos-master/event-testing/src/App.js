import React, { useState } from 'react';

// import Form from './Form';
import FormWithPropsFn from './FormWithPropsFn';

function App() {
  return (
    <FormWithPropsFn submit={({ name }) => { console.log(name) }} />
  );
}

export default App;
