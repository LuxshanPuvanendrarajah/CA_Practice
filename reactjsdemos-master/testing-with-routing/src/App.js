import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import About from './Components/About';
import Header from './Components/Header';
import Home from './Components/Home';

function App() {
  return (
    <>
      <h1>Learn React</h1>
      <Router>
        <Header />
        <Switch>
          <Route path="/" exact>
            <Home />
          </Route>
          <Route path="/about">
            <About />
          </Route>
        </Switch>
      </Router>
    </>
  );
}

export default App;
