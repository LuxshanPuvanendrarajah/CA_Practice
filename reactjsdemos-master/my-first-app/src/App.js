import React from 'react';
import './App.css';
// import ClassComponentWithState from './Components/ClassComponentWithState';
// import ComponentWithUncontrolledInput from './Components/ComponentWithUncontrolledInput';
// import ComponentWithControlledInput from './Components/ComponentWithControlledInput';
// import FormSubmit from './Components/FormSubmit';
import ComponentWithProps from './Components/ComponentWithProps';
// import ComponentWithArrayOfData from './Components/ComponentWithArrayOfData';
// import ComponentConditionalRender from './Components/ComponentConditionalRender';
// import ComponentConditionalTernaryRender from './Components/ComponentConditionalTernaryRender';
// import ComponentConditionalLogicalRender from './Components/ComponentConditionalLogicalRender';
// import ClassComponenentClock from './Components/ClassComponentClock';
// import FunctionComponentClock from './Components/FunctionComponentClock';


function App() {
    const someValue = 10;

    const someDataObject = {
        key1: `value1`,
        key2: 2,
        key3: [`array`, `of`, `strings`]
    };

    const someDataArray = [0, 1, 2, 3, 4];

    const someFunction = () => (`a string from a function`);


    return (
        <React.Fragment>
            <ComponentWithProps
                headerText="Heading text from headerText prop"
                numericProp={"100"}
                expressionProp={100 / 5}
                valueProp={someValue}
                objectProp={someDataObject}
                arrayProp={someDataArray}
                functionProp={someFunction}
                unsuppliedProp="Hello World from parent"
            />
            {/* <ComponentWithArrayOfData /> */}
            {/* <ComponentConditionalRender /> */}
            {/* <ComponentConditionalTernaryRender /> */}
            {/* <ComponentConditionalLogicalRender /> */}
            {/* <ClassComponentWithState /> */}
            {/* <ComponentWithUncontrolledInput /> */}
            {/* <ComponentWithControlledInput /> */}
            {/* <FormSubmit /> */}
            {/* <div className="flex-container">
                <ClassComponenentClock />
                <FunctionComponentClock />
            </div> */}
        </React.Fragment>
    );
}

export default App;
