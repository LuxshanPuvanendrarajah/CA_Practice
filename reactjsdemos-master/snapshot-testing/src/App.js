import React from 'react';
import logo from './logo.svg';
import './App.css';

import ComponentWithProps from './Components/ComponentWithProps';

function App() {
  const props = {
    headerText: `A header`,
    contentText: `Some content`,
    numericProp: 100,
    expressionProp: 2 + 2,
    valueProp: 1000,
    objectProp: {
      key1: `key 1 value`,
      key2: 2,
      key3: [`Key 3`, `array`, `values`]
    },
    arrayProp: [1, 2, 3],
    functionProp: () => `Hello World`,
  }

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React with Ed
        </a>
        <ComponentWithProps {...props} />
      </header>
    </div>
  );
}

export default App;
