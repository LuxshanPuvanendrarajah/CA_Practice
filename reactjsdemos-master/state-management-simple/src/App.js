import React from 'react';
import MyComponent from './MyComponent';
import './App.css';
import DataProvider from './StateManagement/DataProvider';

function App() {
  return (
    <div className="App">
      <DataProvider>
        <MyComponent />
      </DataProvider>
    </div>
  );
}

export default App;
