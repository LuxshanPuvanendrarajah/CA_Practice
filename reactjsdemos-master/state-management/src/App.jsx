import React from 'react';
import { BrowserRouter as Router, NavLink, Switch, Route, Redirect } from 'react-router-dom';
import './App.css';
import FilterableCoursesTable from './Components/FilterableCoursesTable4';
import AddCourseForm from './Components/AddCourseForm2';
import CoursesProvider from './StateManagement/CoursesProvider5';

function App() {
	return (
		<Router>
			<header>
				<nav>
					<ul>
						<li>
							<NavLink to="/">Search</NavLink>
						</li>
						<li>
							<NavLink to="/add">Add</NavLink>
						</li>
						<li>
							<NavLink to="/someunmatchableroute">
								Unmatchable Route
              </NavLink>
						</li>
					</ul>
				</nav>
			</header>

			<Switch>
				<CoursesProvider>
					<Route path="/" exact>
						<FilterableCoursesTable />
					</Route>
					<Route path="/courses">
						<FilterableCoursesTable />
					</Route>
					<Route path="/add">
						<AddCourseForm />
					</Route>
				</CoursesProvider>
				<Route>
					<Redirect to="/" />
				</Route>
			</Switch>

		</Router>
	);
}

export default App;
