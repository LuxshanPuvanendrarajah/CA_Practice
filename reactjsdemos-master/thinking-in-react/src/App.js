import React from 'react';
import './App.css';
import FilterableCoursesTable from './Components/FilterableCoursesTable4';
import AddCourseForm from './Components/AddCourseForm';

function App() {

  return (
    <>
      <FilterableCoursesTable />
      <AddCourseForm />
    </>
  );
}

export default App;
