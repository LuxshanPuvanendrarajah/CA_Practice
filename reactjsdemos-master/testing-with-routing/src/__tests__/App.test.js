import React from 'react';
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import App from '../App';

jest.mock('../Components/About', () => {
  return function About() { return <h1>My Mock About Component</h1> }
});

jest.mock('../Components/Header', () => {
  return function Header() { return <h1>My Mock Header Component</h1> }
});

jest.mock('../Components/Home', () => {
  return function Home() { return <h1>My Mock Home Component</h1> }
});

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('Snapshot renders', () => {
  const appSnapshot = create(<App />).toJSON();

  expect(appSnapshot).toMatchSnapshot();
});
