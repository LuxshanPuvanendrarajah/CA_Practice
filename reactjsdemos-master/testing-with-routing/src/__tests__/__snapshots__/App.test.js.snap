// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
Array [
  <h1>
    Learn React
  </h1>,
  <h1>
    My Mock Header Component
  </h1>,
  <h1>
    My Mock Home Component
  </h1>,
]
`;
